### ISSUE TITLE
HSV: Live feed works but no events are recorded in HSV interface, even when set to Stream & Allow Recording

### ISSUE THREAD

**Issue by eperitus** [seydx/homebridge-camera-ui#336]
**Describe the bug**
Using HSV, camera shows up, live feed works, but nothing is recorded, even when set to Stream & Allow Recording in  Recording Options within HomeKit.  Some other people (ex. quinn2244) are having the same problem on reddit: https://www.reddit.com/r/HomeKit/comments/s2xygx/comment/hshk401/?utm_source=share&utm_medium=web2x&context=3

I've tried with and without audio.  I've waited over 24 hours.  Events show up in the Wyze app interface but not in the HomeKit interface.  The only thing that shows in the HomeKit interface is the live stream.  I have other native HKSV cameras and they work fine with events in the HSV timeline.

**To Reproduce**
Setup Wyze v3 RTSP camera with config file below.

**Expected behavior**
Recording of events in HSV timeline.

**Logs**
No errors in logs.

**Screenshots**
![HSV-livefeed-but-no-events-recorded-in-timeline-1](https://user-images.githubusercontent.com/57232286/149441899-9d7600ed-8acc-44ec-88b0-3c09a554e987.png)

**Environment**
- Node.js Version: v17.3.0
- Homebridge Version: v1.4.0 beta4
- Camera UI Version: v5.06
- Operating System: macOS 12.1
- Browser: N/A
- Config file:

``` 
{
    "name": "CameraUI",
    "debug": false,
    "mqtt": {
        "active": false
    },
    "http": {
        "active": false,
        "localhttp": false
    },
    "smtp": {
        "active": false
    },
    "ftp": {
        "active": false
    },
    "ssl": {
        "active": false
    },
    "options": {
        "videoProcessor": "/opt/homebrew/bin/ffmpeg"
    },
    "cameras": [
        {
            "name": "FrontYardCam",
            "manufacturer": "Wyze",
            "model": "WyzeCam v3",
            "motion": true,
            "unbridge": true,
            "hsv": true,
            "prebufferLength": 4,
            "videoConfig": {
                "source": "-i rtsp://xxx:yyy@192.168.1.105/live",
                "maxStreams": 1,
                "vcodec": "copy",
                "acodec": "libfdk_aac",
                "packetSize": 188,
                "stimeout": 0,
                "audio": true
            },
            "videoanalysis": {
                "active": false
            }
        },
        {
            "prebufferLength": 4,
            "videoConfig": {
                "stimeout": 0
            },
            "videoanalysis": {
                "active": false
            }
        }
    ],
    "platform": "CameraUI"
}
```

**Comment by seydx** [comment]
You need to enable some kind of motion detection. You have several options to pass the motion info from your camera to camera.ui

Without these, camera.ui will not recognize any motion

a) However, if your camera can send am email on motion detection, so enable the SMTP server in the config, and adjust the camera email settings to camera.ui as destination (and camera name as target email adress, eg My+Camera@camera.ui)

b) If your camera supports MQTT notification, enable the MQTT Client in the config json, in your camera block, you can add a "motionMessage" (string or a object) and a "motionTopic" (topic/suffix)

c) If your camera can upload images on motion detection via FTP, enable the FTP Server within config.json. Adjust the camera ftp settings to camera.ui as destination and add as path the camera name

TL:DR

d) If all this doesn't work, and you just want camera.ui to do everything itself, enable "videoanalysis" in config.json and restart HB

**Comment by seydx** [comment]
@eperitus try this config

```json
{
    "name": "CameraUI",
    "debug": false,
    "mqtt": {
        "active": false
    },
    "http": {
        "active": false,
        "localhttp": false
    },
    "smtp": {
        "active": false
    },
    "ftp": {
        "active": false
    },
    "ssl": {
        "active": false
    },
    "options": {
        "videoProcessor": "/opt/homebrew/bin/ffmpeg"
    },
    "cameras": [
        {
            "name": "FrontYardCam",
            "manufacturer": "Wyze",
            "model": "WyzeCam v3",
            "motion": true,
            "unbridge": true,
            "hsv": true,
            "prebufferLength": 4,
            "videoConfig": {
                "source": "-i rtsp://xxx:yyy@192.168.1.105/live",
                "maxStreams": 1,
                "vcodec": "copy",
                "acodec": "libfdk_aac",
                "stimeout": 10,
                "audio": true
            },
            "videoanalysis": {
                "active": true
            }
        }
    ],
    "platform": "CameraUI"
}
```

**Comment by eperitus** [comment]
Thanks very much @seydx , with the config you suggested, I now get this in the log:

[1/13/2022, 7:30:07 PM] [CameraUI] FrontYardCam: Videoanalysis scheduled for restart at 3AM: 450 minutes

This sounds promising.  I'll check again tomorrow to see if any events are recorded in HomeKit.

Where is the video stored while it is awaiting analysis?  Some path on my computer?  Or does it stay on the camera's SD card until camera.ui needs it?  I'm wondering how big it can get at 1080p, if it gets cleaned up (on the computer) after analysis + upload to iCloud, and if I have enough space locally ...

**Comment by seydx** [comment]
with the config above, the hsv recordings will be 20-22s, a video with 1920x1080 will be 2-3mb

To save the hsv recordings also locally, you need to enable recording in the interface

The default port (when not set in config) is 8081 for the interface

So, open http://localhost:8081 (maybe change localhost with ip where camera.ui is running)

Default credentials are:
username: master
password: master

if successfully logged in, go to Settings > Recordings, there you can adjust everything

**Comment by eperitus** [comment]
Just got my first motion event in HSV!  It was about 15 seconds.  A little late capturing a vehicle but it came through.

I don't need to save the recordings locally, just in HSV is fine.  Just wondering if they were analyzed locally and if they were cleaned up.

Didn't know about the 8081 interface - thanks!

**Comment by seydx** [comment]
> 

Enable prebuffering if you want the event before motion has triggered

```
{
    "name": "CameraUI",
    "debug": false,
    "mqtt": {
        "active": false
    },
    "http": {
        "active": false,
        "localhttp": false
    },
    "smtp": {
        "active": false
    },
    "ftp": {
        "active": false
    },
    "ssl": {
        "active": false
    },
    "options": {
        "videoProcessor": "/opt/homebrew/bin/ffmpeg"
    },
    "cameras": [
        {
            "name": "FrontYardCam",
            "manufacturer": "Wyze",
            "model": "WyzeCam v3",
            "motion": true,
            "unbridge": true,
            "hsv": true,
            "prebuffering": true,
            "prebufferLength": 4,
            "videoConfig": {
                "source": "-i rtsp://xxx:yyy@192.168.1.105/live",
                "maxStreams": 1,
                "vcodec": "copy",
                "acodec": "libfdk_aac",
                "stimeout": 10,
                "audio": true
            },
            "videoanalysis": {
                "active": true
            }
        }
    ],
    "platform": "CameraUI"
}
```

**Comment by town3r** [comment]
> Just got my first motion event in HSV! It was about 15 seconds. A little late capturing a vehicle but it came through.
Did you have to modify any of the cam settings w/ in the Wyze app?

**Comment by seydx** [comment]
> Just got my first motion event in HSV! It was about 15 seconds. A little late capturing a vehicle but it came through.
> Did you have to modify any of the cam settings w/ in the Wyze app?

enable "prebuffering" in your config

**Comment by town3r** [comment]
Appologies!
I do have it.

```
        {
            "name": "CameraUI",
            "debug": false,
            "mqtt": {
                "active": false
            },
            "http": {
                "active": false,
                "localhttp": false
            },
            "smtp": {
                "active": false
            },
            "ftp": {
                "active": false
            },
            "ssl": {
                "active": false
            },
            "cameras": [
                {
                    "name": "ToWn3r-WyzeCamv3",
                    "manufacturer": "Wyze Labs, Inc.",
                    "model": "WyzeCam v3",
                    "motion": true,
                    "unbridge": true,
                    "hsv": true,
                    "prebuffering": true,
                    "prebufferLength": 4,
                    "videoConfig": {
                        "source": "-i rtsp://username:password@192.168.1.182/live",
                        "maxStreams": 1,
                        "vcodec": "copy",
                        "acodec": "libfdk_aac",
                        "stimeout": 10,
                        "audio": false
                    },
                    "videoanalysis": {
                        "active": true
                    }
                }
            ],
            "platform": "CameraUI"
```

**Comment by seydx** [comment]
> Appologies! I do have it.
> 
> ```
>         {
>             "name": "CameraUI",
>             "debug": false,
>             "mqtt": {
>                 "active": false
>             },
>             "http": {
>                 "active": false,
>                 "localhttp": false
>             },
>             "smtp": {
>                 "active": false
>             },
>             "ftp": {
>                 "active": false
>             },
>             "ssl": {
>                 "active": false
>             },
>             "cameras": [
>                 {
>                     "name": "ToWn3r-WyzeCamv3",
>                     "manufacturer": "Wyze Labs, Inc.",
>                     "model": "WyzeCam v3",
>                     "motion": true,
>                     "unbridge": true,
>                     "hsv": true,
>                     "prebuffering": true,
>                     "prebufferLength": 4,
>                     "videoConfig": {
>                         "source": "-i rtsp://username:password@192.168.1.182/live",
>                         "maxStreams": 1,
>                         "vcodec": "copy",
>                         "acodec": "libfdk_aac",
>                         "stimeout": 10,
>                         "audio": false
>                     },
>                     "videoanalysis": {
>                         "active": true
>                     }
>                 }
>             ],
>             "platform": "CameraUI"
> ```

What does not work with it?

**Comment by eperitus** [comment]
I enabled prebuffering and didn't modify any settings in the Wyze app.  I am using the rest of the config above but the HSV interface is showing fewer events than the Wyze camera app.  

For example, I have 3 events total in HSV in the past 24 hours, and well over 30 events in Wyze.

I see a lot of this in the log:
[1/14/2022, 2:00:31 PM] [CameraUI] FrontYardCam: Skip saving the recording, the file might be corrupted because the recording process was cancelled by HSV
[1/14/2022, 2:00:38 PM] [CameraUI] FrontYardCam: Motion OFF - Motion handler timeout.
[1/14/2022, 2:03:19 PM] [CameraUI] FrontYardCam: Motion OFF
[1/14/2022, 2:10:05 PM] [CameraUI] FrontYardCam: Motion ON
[1/14/2022, 2:10:12 PM] [CameraUI] FrontYardCam: Closing recording process

I also see "Camera Not Available - Wait until someone else in this home stops viewing this camera and try again" in the Home app for the Wyze camera - but nobody else is viewing it.

Should I open a new issue for this or is OK to keep it in this thread?

**Comment by town3r** [comment]
No motion is detected so nothing is recorded to HSV

tried adding and removing the camera from homekit being i originally had it added via `Homebridge Camera FFmpeg` and just copied it's config values over to Homebridge Camera Ui followed by removing it from Camera FFmpeg.

Didn't have to re-add it. So i did just to be safe.

**Comment by seydx** [comment]
> I enabled prebuffering and didn't modify any settings in the Wyze app. I am using the rest of the config above but the HSV interface is showing fewer events than the Wyze camera app.
> 
> For example, I have 3 events total in HSV in the past 24 hours, and well over 30 events in Wyze.
> 
> I see a lot of this in the log: [1/14/2022, 2:00:31 PM] [CameraUI] FrontYardCam: Skip saving the recording, the file might be corrupted because the recording process was cancelled by HSV [1/14/2022, 2:00:38 PM] [CameraUI] FrontYardCam: Motion OFF - Motion handler timeout. [1/14/2022, 2:03:19 PM] [CameraUI] FrontYardCam: Motion OFF [1/14/2022, 2:10:05 PM] [CameraUI] FrontYardCam: Motion ON [1/14/2022, 2:10:12 PM] [CameraUI] FrontYardCam: Closing recording process
> 
> I also see "Camera Not Available - Wait until someone else in this home stops viewing this camera and try again" in the Home app for the Wyze camera - but nobody else is viewing it.
> 
> Should I open a new issue for this or is OK to keep it in this thread?

Open Interface > Settings > Cameras > Select Camera > Videoanalysis > Adjust sensitivity (lower should result in more events

camera.ui has a "persistent" connection to the camera (when prebuffering is active), I assume that is what is meant by this

**Comment by seydx** [comment]
> No motion is detected so nothing is recorded to HSV
> 
> tried adding and removing the camera from homekit being i originally had it added via `Homebridge Camera FFmpeg` and just copied it's config values over to Homebridge Camera Ui followed by removing it from Camera FFmpeg.
> 
> Didn't have to re-add it. So i did just to be safe.

Restart HB, trigger motion, post the log here pls

**Comment by eperitus** [comment]
I disabled prebuffering (this worked to alleviate the camera not available problem) and decreased the sensitivity from 50 down to 20.  No new events yet but I'll keep checking.  Thanks again seydx!

**Comment by seydx** [comment]
> I disabled prebuffering (this worked to alleviate the camera not available problem) and decreased the sensitivity from 50 down to 20. No new events yet but I'll keep checking. Thanks again seydx!

The next version v5.0.7 (still under development) will let you to customize the videoanalysis process more through the ui, and also the timer for retrigger a new motion is lower, may this will fix it for you

but till then keep checking your log

**Comment by town3r** [comment]
> > No motion is detected so nothing is recorded to HSV
> > tried adding and removing the camera from homekit being i originally had it added via `Homebridge Camera FFmpeg` and just copied it's config values over to Homebridge Camera Ui followed by removing it from Camera FFmpeg.
> > Didn't have to re-add it. So i did just to be safe.
> 
> Restart HB, trigger motion, post the log here pls

By Trigger Motion do you mean just dance in front of the camera and see if i get something like:
I don't get anything liek this: [1/14/2022, 2:10:05 PM] [CameraUI] FrontYardCam: Motion ON



```
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Prebuffering scheduled for restart at 2AM: 556 minutes
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Incoming ping request for: rtsp://username:password@192.168.1.182/live - Timeout: 1s
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Pinging 192.168.1.182:80
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Pinging 192.168.1.182:80 - successful
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Start videoanalysis...
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Videoanalysis command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -hide_banner -loglevel error -hwaccel auto -analyzeduration 0 -probesize 500000 -f mpegts -i tcp://127.0.0.1:27064 -an -vcodec pam -pix_fmt rgb24 -f image2pipe -vf fps=2,scale=640:360 pipe:1
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Videoanalysis: Sensitivity: 25 - Active zones: []
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Videoanalysis scheduled for restart at 3AM: 616 minutes
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Configuring unbridged accessory...
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Setting up accessory...
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Adding motion sensor service
[14/01/2022, 16:44:20] [CameraUI] ToWn3r-WyzeCamv3: Initializing HomeKit Secure Video
[14/01/2022, 16:44:20] [CameraUI] camera.ui v5.0.6 is listening on port 8081 (http)
[14/01/2022, 16:44:20] ToWn3r-WyzeCamv3 8736 is running on port 46689.
```

**Comment by town3r** [comment]
`[14/01/2022, 16:46:49] [CameraUI] ToWn3r-WyzeCamv3: Snapshot requested: 1280 x 720
[14/01/2022, 16:46:49] [CameraUI] ToWn3r-WyzeCamv3: Snapshot command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -hide_banner -loglevel error -analyzeduration 0 -probesize 731696 -f mpegts -i tcp://127.0.0.1:33782 -frames:v 1 -f image2 -
[14/01/2022, 16:46:50] [CameraUI] ToWn3r-WyzeCamv3: Prebuffer request ended
[14/01/2022, 16:46:50] [CameraUI] ToWn3r-WyzeCamv3: Fetching snapshot took 1.465 seconds.
[14/01/2022, 16:46:50] [CameraUI] ToWn3r-WyzeCamv3: Sending snapshot: 1280x720
[14/01/2022, 16:46:50] [CameraUI] ToWn3r-WyzeCamv3: Resize command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -i pipe: -frames:v 1 -filter:v scale='min(1280,iw)':'min(720,ih)':force_original_aspect_ratio=decrease -f image2 -
[14/01/2022, 16:46:51] [CameraUI] ToWn3r-WyzeCamv3: Start stream requested: 1280x720, 30 fps, 299 kbps
[14/01/2022, 16:46:51] [CameraUI] ToWn3r-WyzeCamv3: Incoming ping request for: rtsp://username:password@192.168.1.182/live - Timeout: 1s
[14/01/2022, 16:46:51] [CameraUI] ToWn3r-WyzeCamv3: Pinging 192.168.1.182:80
[14/01/2022, 16:46:51] [CameraUI] ToWn3r-WyzeCamv3: Pinging 192.168.1.182:80 - successful
[14/01/2022, 16:46:51] [CameraUI] ToWn3r-WyzeCamv3: Setting prebuffer stream as input
[14/01/2022, 16:46:51] [CameraUI] ToWn3r-WyzeCamv3: Starting video stream: native
[14/01/2022, 16:46:51] [CameraUI] ToWn3r-WyzeCamv3: Stream command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -hide_banner -loglevel level -analyzeduration 0 -probesize 740156 -f mpegts -i tcp://127.0.0.1:32596 -an -sn -dn -vcodec copy -pix_fmt yuv420p -color_range mpeg -f rawvideo -payload_type 99 -ssrc 2148188 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params qRsOnxhpyC1Cu4BXdoqNWBqRojl08Od9nKgaaBoO srtp://192.168.1.7:59192?rtcpport=59192&pkt_size=1318 -progress pipe:1
[14/01/2022, 16:46:52] [CameraUI] ToWn3r-WyzeCamv3: Getting the first frames took 1.545 seconds.
[14/01/2022, 16:46:55] [CameraUI] ToWn3r-WyzeCamv3: Stop stream requested
[14/01/2022, 16:46:55] [CameraUI] ToWn3r-WyzeCamv3: Stopped video stream.
[14/01/2022, 16:46:55] [CameraUI] ToWn3r-WyzeCamv3: Prebuffer request ended
[14/01/2022, 16:46:55] [CameraUI] ToWn3r-WyzeCamv3: FFmpeg exited with code: null and signal: SIGKILL (Expected)
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Motion detected via Videoanalysis: {"name":"all","diffs":206637,"total":230400,"percent":89.68619537353516}
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Motion ON
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Handling through extern controller..
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Video fragments requested from HSV
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Setting prebuffer stream as input
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Prebuffer requested with a duration of -4s
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Start recording...
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Recording command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -hide_banner -analyzeduration 0 -probesize 500000 -f mp4 -i tcp://127.0.0.1:23801 -f mp4 -vcodec copy -an -movflags frag_keyframe+empty_moov+default_base_moof -max_muxing_queue_size 1024 tcp://127.0.0.1:12982
[14/01/2022, 16:47:42] [CameraUI] ToWn3r-WyzeCamv3: Recording started
[14/01/2022, 16:47:57] [CameraUI] ToWn3r-WyzeCamv3: Motion OFF - Motion handler timeout.
[14/01/2022, 16:48:00] [CameraUI] ToWn3r-WyzeCamv3: Recording completed (HSV)
[14/01/2022, 16:48:00] [CameraUI] ToWn3r-WyzeCamv3: New custom alert
[14/01/2022, 16:48:00] [CameraUI] ToWn3r-WyzeCamv3: Notifications not enabled, skip Webhook..
[14/01/2022, 16:48:00] [CameraUI] ToWn3r-WyzeCamv3: Notifications not enabled, skip Alexa..
[14/01/2022, 16:48:00] [CameraUI] ToWn3r-WyzeCamv3: Recording not enabled, skip recording..
[14/01/2022, 16:48:00] [CameraUI] ToWn3r-WyzeCamv3: Recording skipped. Notification skipped.
[14/01/2022, 16:48:01] [CameraUI] ToWn3r-WyzeCamv3: Closing recording process
[14/01/2022, 16:48:01] [CameraUI] ToWn3r-WyzeCamv3: Prebuffer request ended
[14/01/2022, 16:48:01] [CameraUI] ToWn3r-WyzeCamv3: FFmpeg recording process exited (expected)
[14/01/2022, 16:49:12] [CameraUI] ToWn3r-WyzeCamv3: Motion OFF
[14/01/2022, 16:49:12] [CameraUI] ToWn3r-WyzeCamv3: Handling through extern controller..
[14/01/2022, 16:53:53] [CameraUI] ToWn3r-WyzeCamv3: Snapshot requested: 1280 x 720
[14/01/2022, 16:53:53] [CameraUI] ToWn3r-WyzeCamv3: Snapshot command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -hide_banner -loglevel error -analyzeduration 0 -probesize 599156 -f mpegts -i tcp://127.0.0.1:31980 -frames:v 1 -f image2 -
[14/01/2022, 16:53:54] [CameraUI] ToWn3r-WyzeCamv3: Prebuffer request ended
[14/01/2022, 16:53:54] [CameraUI] ToWn3r-WyzeCamv3: Fetching snapshot took 0.618 seconds.
[14/01/2022, 16:53:54] [CameraUI] ToWn3r-WyzeCamv3: Sending snapshot: 1280x720
[14/01/2022, 16:53:54] [CameraUI] ToWn3r-WyzeCamv3: Resize command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -i pipe: -frames:v 1 -filter:v scale='min(1280,iw)':'min(720,ih)':force_original_aspect_ratio=decrease -f image2 -
[14/01/2022, 16:54:04] [CameraUI] ToWn3r-WyzeCamv3: Snapshot requested: 1280 x 720
[14/01/2022, 16:54:04] [CameraUI] ToWn3r-WyzeCamv3: Snapshot command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -hide_banner -loglevel error -analyzeduration 0 -probesize 561180 -f mpegts -i tcp://127.0.0.1:25931 -frames:v 1 -f image2 -
[14/01/2022, 16:54:06] [CameraUI] ToWn3r-WyzeCamv3: Prebuffer request ended
[14/01/2022, 16:54:06] [CameraUI] ToWn3r-WyzeCamv3: Fetching snapshot took 1.39 seconds.
[14/01/2022, 16:54:06] [CameraUI] ToWn3r-WyzeCamv3: Sending snapshot: 1280x720
[14/01/2022, 16:54:06] [CameraUI] ToWn3r-WyzeCamv3: Resize command: /usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg -i pipe: -frames:v 1 -filter:v scale='min(1280,iw)':'min(720,ih)':force_original_aspect_ratio=decrease -f image2 -`

**Comment by town3r** [comment]
ohh! looks like its starting to record things

**Comment by town3r** [comment]
@eperitus did the home app notify you of motion?

is it detecting people or animals yet?

**Comment by eperitus** [comment]
Not yet, and I walked by it 3 different times.  HSV detected 3 vehicles overnight but nothing since this morning or afternoon, and there have been plenty of people and vehicles going by.  So I'm not sure the sensitivity setting is working.  Meanwhile, the Wyze app is showing all of the events, and my other native HomeKit cameras are reporting events in HSV.

I just restarted homebridge and will see if any new events occur.

**Comment by town3r** [comment]
Videoanalysis is all done locally correct? it's not being sent to like your server, right? 😅

**Comment by seydx** [comment]
> Videoanalysis is all done locally correct? it's not being sent to like your server, right? 😅

everything is done locally, also your database for camera.ui which contains all information like user credentials, recordings etc stored locally

**Comment by town3r** [comment]
Figured as much. :) Thanks for confirming.

Installing the v5.0.7 beta 1.....jk

`USER: pi
DIR: /usr/local
CMD: sudo -E -n npm install -g homebridge-camera-ui@5.0.7-beta.1

npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm ERR! code ERR_SSL_DECRYPTION_FAILED_OR_BAD_RECORD_MAC
npm ERR! 3070130688:error:1408F119:SSL routines:ssl3_get_record:decryption failed or bad record mac:../deps/openssl/openssl/ssl/record/ssl3_record.c:677:
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2022-01-14T23_58_24_033Z-debug-0.log

Command failed. Please review log for details.`

**Comment by seydx** [comment]
> Figured as much. :) Thanks for confirming.
> 
> Installing the v5.0.7 beta 1.....jk
> 
> ```
> USER: pi
> DIR: /usr/local
> CMD: sudo -E -n npm install -g homebridge-camera-ui@5.0.7-beta.1
> 
> npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
> npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
> npm ERR! code ERR_SSL_DECRYPTION_FAILED_OR_BAD_RECORD_MAC
> npm ERR! 3070130688:error:1408F119:SSL routines:ssl3_get_record:decryption failed or bad record mac:../deps/openssl/openssl/ssl/record/ssl3_record.c:677:
> npm ERR! 
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/pi/.npm/_logs/2022-01-14T23_58_24_033Z-debug-0.log
> 
> Command failed. Please review log for details.
> ```

Oh, were you running v5.0.6 before without any problems? Which node version?

**Comment by town3r** [comment]
Node v16.13.2
Npm v8.3.1

**Comment by town3r** [comment]
False alarm.
Third attempt seems to have worked.

**Comment by seydx** [comment]
v5.0.7-beta.3 is online with better videoanalysis and several bugfixes

`sudo npm i -g homebridge-camera-ui@5.0.7-beta.3`

**Comment by wiggins-philip** [comment]
I'm getting this message but recordings still don't show up. I've tried hsv and with the interface. This is the output with recording using the interface, but there are no recordings when I go to find them. 

`VIDEOANALYSIS: New message: Data: [{"zone":"region0","percent":49.54426956176758,"sensitivity":51,"dwell":60,"forceClose":3}] - Motion: detected - Camera: Office Cam
[1/24/2023, 10:24:42 AM] [CameraUI] Office Cam: Skip motion trigger. At Home is active and Office Cam is not excluded!
[1/24/2023, 10:24:44 AM] [CameraUI] Office Cam: New motion detected, resetting motion in 60s..
[1/24/2023, 10:24:44 AM] [CameraUI] Office Cam: Motion data: [{"zone":"region0","percent":42.17881774902344,"sensitivity":59,"dwell":60,"forceClose":3}]}
[1/24/2023, 10:24:46 AM] [CameraUI] Office Cam: New motion detected, resetting motion in 60s..
[1/24/2023, 10:24:46 AM] [CameraUI] Office Cam: Motion data: [{"zone":"region0","percent":46.7890625,"sensitivity":54,"dwell":60,"forceClose":3}]}
[1/24/2023, 10:25:00 AM] [Home Alarm] SSAPI socket `heartbeat`
[1/24/2023, 10:25:03 AM] [CameraUI] VIDEOANALYSIS: New message: Data: {"time":"2023-01-24T16:25:03.762Z","event":"dwellTime (60s)"} - Motion: resetted - Camera: Kitty Cam
[1/24/2023, 10:25:03 AM] [CameraUI] Kitty Cam: Skip motion trigger. At Home is active and Kitty Cam is not excluded!`
